### PR TITLE
fix issue: eventData deserialization fails when the payload size is >1000bytes

### PR DESCRIPTION
**Pull request by SreeramGarlapati**
fix for issue #70
Thanks a lot @CodingCat for filing the issue and identifying the exact problem.

### CODE REVIEW

File: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventData.java [hunk 1]
```diff
@@ -322,18 +323,22 @@ private void writeObject(ObjectOutputStream out) throws IOException
 	{
 		out.defaultWriteObject();
 		
-		out.writeInt(this.bodyData.getLength());
-		out.write(this.bodyData.getArray(), this.bodyData.getArrayOffset(), this.bodyData.getLength());
+                out.writeInt(this.bodyData == null ? BODY_DATA_NULL : this.bodyData.getLength());
```
Comment (sjkwak) [inline azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventData.java]: >this.bodyData == null ? BODY_DATA_NULL : [](start = 29, length = 40)

Why do we need this? this.bodyData can't be null, right?

Comment (SreeramGarlapati) [inline azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventData.java]: For AmqpMessage sent using open libraries - body can be null.

File: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventData.java [hunk 2]
```diff
@@ -322,18 +323,22 @@ private void writeObject(ObjectOutputStream out) throws IOException
...
 		in.defaultReadObject();
 		
 		final int length = in.readInt();
-		final byte[] data = new byte[length];
-		in.read(data, 0, length);
-		this.bodyData = new Binary(data, 0, length);
+		if (length != BODY_DATA_NULL) {
+                    
+                    final byte[] data = new byte[length];
+                    in.readFully(data, 0, length);
```
Comment (sjkwak) [inline azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventData.java]: >in.readFully( [](start = 20, length = 13)

Do we actually validate body data in our stress tests? We have CRC check for .NET cases, we should have the same tests ported to Java.

Comment (SreeramGarlapati) [inline azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventData.java]: In general, having extra level of validation is definitely good idea.
Just to be on same page - For this specific PR - this change wouldn't benefit from that though... this desrialization is for non-amqp - serialization code paths - like, for ex: if customers want to persist EventData to a file.

**Pull request merged**